src/src/test/java/StringManipulationTest.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StringManipulationTest {
+
+    private StringManipulationInterface manipulatedstring;
+
+    @BeforeEach
+    public void setUp() {
+        manipulatedstring = new StringManipulation();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        manipulatedstring = null;
+    }
+
+    @Test
+    public void testCount1() {
+        manipulatedstring.setString("This is my string");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
+    }
+
+    @Test
+    public void testCount2() {
+        manipulatedstring.setString("One more string for testing");
+        int length = manipulatedstring.count();
+        assertEquals(5, length);
+    }
+
+    @Test
+    public void testCount3() {
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    @Test
+    public void testCount4() {
+        manipulatedstring.setString("SingleWord");
+        int length = manipulatedstring.count();
+        assertEquals(1, length);
+    }
+
```

Comment:
You could add some tests to handle if the string is empty or if after removing you end up with an empty string